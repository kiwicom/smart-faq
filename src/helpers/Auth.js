// @flow
import Cookies from 'js-cookie';
import { Requester } from './Requests';

export const COOKIE_LOGIN_KEY = 'ua_session_token';
export const getSessionToken = () => Cookies.get(COOKIE_LOGIN_KEY);

export async function doLogin(email: string, password: string) {
  try {
    const token = await Requester.login(email, password);
    Cookies.set(COOKIE_LOGIN_KEY, token, { path: '/', domain: null });
    return token;
  } catch (error) {
    console.error(error.message); //eslint-disable-line
    return '';
  }
}

export async function socialLogin(provider: string) {
  const location = window.location.href.replace(
    /((utm_[a-z]+)|(session_[a-z]+))=[^&]+(&|$)/g,
    '',
  );
  const redirectUrl = `${location}${
    location.includes('?') ? '&' : '?'
  }oauth-login=true`
    .replace(/,/g, '%2C')
    .replace(/~/g, '%7E');

  const USER_APP_ID = process.env.KIWILOGIN_USER;
  const payload = {
    provider_id: provider,
    app_id: USER_APP_ID,
    redirect_url: redirectUrl,
  };
  try {
    return await Requester.socialLogin(payload);
  } catch (error) {
    console.error(error.message); //eslint-disable-line
    return;
  }
}
