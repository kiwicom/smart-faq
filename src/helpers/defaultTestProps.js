// @flow
export const user = {
  id: '3',
  email: 'email@kiwi.com',
  firstname: 'MyName',
  lastname: 'This is my name',
};
