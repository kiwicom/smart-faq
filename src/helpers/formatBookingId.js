// @flow

const formatBookingId = (bookingDatabaseId: ?number): string => {
  const splitBookingId: string[] =
    String(bookingDatabaseId)
      .split('')
      .reverse()
      .join('')
      .match(/.{1,3}/g) || [];

  return splitBookingId
    .join(' ')
    .split('')
    .reverse()
    .join('');
};

export default formatBookingId;
