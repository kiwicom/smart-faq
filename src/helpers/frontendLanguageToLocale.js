// @flow

export const DEFAULT_LOCALE = 'en_GB';

export default {
  en: 'en_GB',
  us: 'en_US',
  cz: 'cs_CZ',
};

export const parseDashToUnderscore = (locale: string): string =>
  locale.replace('-', '_');
