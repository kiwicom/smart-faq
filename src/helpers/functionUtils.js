// @flow

export const debounce = (
  fn: any => any,
  timeout: number = 250,
  interval?: TimeoutID,
) => (...args: any) => {
  clearTimeout(interval);
  interval = setTimeout(() => fn(...args), timeout);
};
