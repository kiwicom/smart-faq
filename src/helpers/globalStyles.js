// @flow

export const animationStyles = (delayTime: number) => `
  @keyframes enterLeft {
    0% {
      -webkit-transform: translateX(400px);
      transform: translateX(400px);
      will-change: transform;
    }
    100% {
      -webkit-transform: translateX(0px);
      transform: translateX(0px);
      will-change: transform;
    }
  }
  @keyframes enterRight {
    0% {
      -webkit-transform: translateX(-400px);
      transform: translateX(-400px);
      will-change: transform;
    }
    100% {
      -webkit-transform: translateX(0px);
      transform: translateX(0px);
      will-change: transform;
    }
  }

  @keyframes exitLeft {
    0% {
      opacity: 1;
      -webkit-transform: translateX(0px);
      transform: translateX(0px);
      will-change: transform;
    }
    100% {
      opacity: 0;
      -webkit-transform: translateX(-400px);
      transform: translateX(-400px);
      will-change: transform;
    }
  }
  @keyframes exitRight {
    0% {
      opacity: 1;
      -webkit-transform: translateX(0px);
      transform: translateX(0px);
      will-change: transform;
    }
    100% {
      opacity: 0;
      -webkit-transform: translateX(400px);
      transform: translateX(400px);
      will-change: transform;
    }
  }

  .SFAQ-TR-slideLeft-enter {
    animation: enterLeft ${delayTime}ms ease-in-out;
  }

  .SFAQ-TR-slideLeft-exit {
    animation: exitLeft ${delayTime}ms ease-in-out;
  }

  .SFAQ-TR-slideRight-enter {
    animation: enterRight ${delayTime}ms ease-in-out;
  }

  .SFAQ-TR-slideRight-exit {
    animation: exitRight ${delayTime}ms ease-in-out;
  }
  .transition-container {
    position: relative;
  }
  .transition-container div.static-faq {
    top: 0;
    left: 0;
    position: absolute;
  }
`;

export const delayTime = 350;
export const transitionStyles = animationStyles(delayTime);
