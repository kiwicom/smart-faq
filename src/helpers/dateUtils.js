// @flow

import * as React from 'react';

import { LanguageContext } from '../context/Language';
import frontendLanguageToLocale, {
  DEFAULT_LOCALE,
} from './frontendLanguageToLocale';

export const URGENCY_THRESHOLD = 48;

type Props = {|
  dateString: string,
|};

export const FormatDate = ({ dateString }: Props) => {
  const options = {
    weekday: 'short',
    month: 'short',
    day: 'numeric',
    timeZone: 'UTC',
  };

  return (
    <LanguageContext.Consumer>
      {language => {
        const locale = (
          frontendLanguageToLocale[language] || DEFAULT_LOCALE
        ).replace('_', '-');

        return (
          <div>{new Date(dateString).toLocaleDateString(locale, options)}</div>
        );
      }}
    </LanguageContext.Consumer>
  );
};

export const formatDepartureDate = (dateString: string) => {
  const dateUTC = new Date(dateString);
  const month = dateUTC.getUTCMonth() + 1;
  const day = dateUTC.getUTCDate();
  const year = dateUTC.getUTCFullYear();
  return `${day}/${month}/${year}`;
};

export const formatCountDown = (hoursLeft: number, __t: string => string) => {
  const nhours = Math.floor(hoursLeft);
  const nmins = Math.floor((hoursLeft - nhours) * 60);
  const ndays = Math.floor(hoursLeft / 24);
  if (hoursLeft < URGENCY_THRESHOLD) {
    return nmins
      ? `${nhours}${__t('TimeUnits.Abbr.Hours')} ${nmins}${__t(
          'TimeUnits.Abbr.Minutes',
        )}`
      : `${nhours}${__t('TimeUnits.Abbr.Hours')}`;
  }
  return `${ndays} ${__t('TimeUnits.Days')}`;
};

export const formatTimeDuration = (mins: number) => {
  const nhours = Math.floor(mins / 60);
  const nmins = mins - nhours * 60;
  return nhours ? `${nhours}h ${nmins}m` : `${nmins}m`;
};

export const formatHour = (date: string) => {
  const options = { hour: 'numeric', timeZone: 'UTC', minute: 'numeric' };
  return new Date(date).toLocaleString('en-US', options);
};
