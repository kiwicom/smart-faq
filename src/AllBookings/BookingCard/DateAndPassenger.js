// @flow

import idx from 'idx';
import * as React from 'react';
import { graphql, createFragmentContainer } from 'react-relay';
import { Text } from '@kiwicom/orbit-components';
import { Passengers } from '@kiwicom/orbit-components/lib/icons';
import Trans from '@kiwicom/nitro/lib/components/Text';

import bookingStatuses from '../../common/booking/bookingStatuses';
import { formatDepartureDate } from '../../helpers/dateUtils';
import type { DateAndPassenger_departure } from './__generated__/DateAndPassenger_departure.graphql';
import type { DateAndPassenger_booking } from './__generated__/DateAndPassenger_booking.graphql';

type Props = {
  departure: DateAndPassenger_departure,
  booking: DateAndPassenger_booking,
};

export const DateAndPassenger = (props: Props) => {
  const departureDate = idx(props.departure, _ => _.time) || '';
  const bookingDate = idx(props.booking, _ => _.bookingDate) || '';
  const passengerCount = idx(props.booking, _ => _.passengerCount) || 0;
  const status =
    (props.booking.status && bookingStatuses()[props.booking.status]) || {};

  return (
    <div className="fields">
      <div className="section">
        <div className="label">
          <Text type="secondary" size="small">
            <Trans
              t={__('smartfaq.all_bookings.booking_card.departure_date')}
            />
          </Text>
        </div>
        <Text>{formatDepartureDate(departureDate)}</Text>
      </div>
      <div className="section">
        <div className="label">
          <Text type="secondary" size="small">
            <Trans t={__('smartfaq.all_bookings.booking_card.booking_date')} />
          </Text>
        </div>
        <Text>{formatDepartureDate(bookingDate)}</Text>
      </div>
      <div className="section">
        <div className="label">
          <Text type="secondary" size="small">
            <Trans t={__('smartfaq.all_bookings.booking_card.includes')} />
          </Text>
        </div>
        <Text>
          <Passengers size="small" customColor="#bac7d5" />
          {passengerCount}
        </Text>
      </div>
      <div className="section">
        <div className="label">
          <Text type="secondary" size="small">
            <Trans t={__('smartfaq.all_bookings.booking_card.status')} />
          </Text>
        </div>
        <div style={{ color: status.color, fontSize: '14px' }}>
          {status.text}
        </div>
      </div>
    </div>
  );
};

export default createFragmentContainer(
  DateAndPassenger,
  graphql`
    fragment DateAndPassenger_booking on BookingInterface {
      status
      passengerCount
      bookingDate
    }

    fragment DateAndPassenger_departure on RouteStop {
      time
    }
  `,
);
