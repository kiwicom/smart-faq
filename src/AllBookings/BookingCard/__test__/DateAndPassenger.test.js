// @flow

import * as React from 'react';
import { shallow, render } from 'enzyme';
import ThemeProvider from '@kiwicom/orbit-components/lib/Theming/ThemeProvider';

import { DateAndPassenger } from '../DateAndPassenger';
import { formatDepartureDate } from '../../../helpers/dateUtils';

const mockRefType: any = null;
const departure = {
  time: '2017-11-23T20:40:00.000Z',
  $refType: mockRefType,
};
const booking = {
  status: 'CLOSED',
  passengerCount: 1,
  bookingDate: '2018-05-14',
  $refType: mockRefType,
};
const translate = () => 'Closed';

describe('DateAndPassenger', () => {
  it('should render correctly', () => {
    expect(
      shallow(
        <ThemeProvider>
          <DateAndPassenger
            departure={departure}
            t={translate}
            booking={booking}
          />
        </ThemeProvider>,
      ),
    ).toMatchSnapshot();
  });

  it('should render Passengers icon', () => {
    const component = shallow(
      <DateAndPassenger
        departure={departure}
        t={translate}
        booking={booking}
      />,
    );
    expect(component.find('Passengers').exists()).toBeTruthy();
  });

  it('should show correct date format', () => {
    const wrapper = render(
      <ThemeProvider>
        <DateAndPassenger
          departure={departure}
          t={translate}
          booking={booking}
        />
      </ThemeProvider>,
    );
    const time = departure.time;
    const formatedDate = formatDepartureDate(time);
    expect(wrapper.text()).toContain(formatedDate);
  });
});
