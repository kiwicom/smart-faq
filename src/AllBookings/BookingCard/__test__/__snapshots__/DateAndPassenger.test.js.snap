// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DateAndPassenger should render correctly 1`] = `
<div
  className="fields"
>
  <div
    className="section"
  >
    <div
      className="label"
    >
      <Text
        size="small"
        type="secondary"
      >
        <Text
          html={false}
          t="smartfaq.all_bookings.booking_card.departure_date"
          values={Object {}}
        />
      </Text>
    </div>
    <Text>
      23/11/2017
    </Text>
  </div>
  <div
    className="section"
  >
    <div
      className="label"
    >
      <Text
        size="small"
        type="secondary"
      >
        <Text
          html={false}
          t="smartfaq.all_bookings.booking_card.booking_date"
          values={Object {}}
        />
      </Text>
    </div>
    <Text>
      14/5/2018
    </Text>
  </div>
  <div
    className="section"
  >
    <div
      className="label"
    >
      <Text
        size="small"
        type="secondary"
      >
        <Text
          html={false}
          t="smartfaq.all_bookings.booking_card.includes"
          values={Object {}}
        />
      </Text>
    </div>
    <Text>
      <Passengers
        customColor="#bac7d5"
        size="small"
      />
      1
    </Text>
  </div>
  <div
    className="section"
  >
    <div
      className="label"
    >
      <Text
        size="small"
        type="secondary"
      >
        <Text
          html={false}
          t="smartfaq.all_bookings.booking_card.status"
          values={Object {}}
        />
      </Text>
    </div>
    <div
      style={
        Object {
          "color": "#ffc345",
          "fontSize": "14px",
        }
      }
    >
      Closed
    </div>
  </div>
</div>
`;
